tests/group_functions/group_scan.h

```diff
@@ -303,6 +303,11 @@ void check_scan_over_group(sycl::queue& queue, sycl::range<D> range, OpT op,
   bool ret_type_e = false;
   bool ret_type_i = false;
 
+  if (((range_size * (range_size + 1) / 2) + T(init)) >
+      std::numeric_limits<T>::max()) {
+    return;
+  }
+
```

Comment:
I still think that silently skipping checks is a bad idea. Please, replace with an assertion.